modules/mixins/cacheMixin.js

```diff
@@ -185,6 +185,22 @@
            */
           requesterDispose: function() {
             parentInstance.removeRequester(ownerKey);
+          },
+
+          /**
+           * In addition to removing the model from the collection also remove it from the list of tracked ids.
+           * @param modelIdentifier {*} same duck-typing as Backbone.Collection.get():
+           *                              by id, cid, model object with id or cid properties,
+           *                              or an attributes object that is transformed through modelId
+           */
+          remove: function(modelIdentifier) {
+            var model = this.get(modelIdentifier);
+            parentClass.remove.apply(this, arguments);
+            if (model) {
+              var trackedIdsWithoutModel = this.getTrackedIds();
+              trackedIdsWithoutModel = _.without(trackedIdsWithoutModel, model.id);
```

Comment:
tracked ids are always model.id right? never by some other value

Reply:
yes that is how the cache collection is setup?  Let me look closer.

Reply:
addModelAndTrack() uses model.id, but you could technically use the model's cid (although you'd be sending that to the server which would be weird) or an object with an id property (or property matching the idAttribute of the model).  Note model.id is already normalized for idAttribute value (equivalent to model.get(model.idAttribute)).